### ISSUE TITLE
[easy] Fix macro redefined warnings in C code

### ISSUE THREAD

**Issue by aantron** [ocsigen/lwt#426]
After applying clang-format to the C code (#416, part of #393), we get these warnings:

```
In file included from lwt_unix_stubs.c:98:
./lwt_unix_unix.h:839:9: warning: 'GET_INET_ADDR' macro redefined [-Wmacro-redefined]
#define GET_INET_ADDR(v) (*((struct in_addr *)(v)))
        ^
/Users/antron/.opam/4.04-lwt/lib/ocaml/caml/socketaddr.h:51:9: note: previous definition is here
#define GET_INET_ADDR(v) (*((struct in_addr *) (v)))
        ^
In file included from lwt_unix_stubs.c:98:
./lwt_unix_unix.h:2365:9: warning: 'GET_INET6_ADDR' macro redefined [-Wmacro-redefined]
#define GET_INET6_ADDR(v) (*((struct in6_addr *)(v)))
        ^
/Users/antron/.opam/4.04-lwt/lib/ocaml/caml/socketaddr.h:55:9: note: previous definition is here
#define GET_INET6_ADDR(v) (*((struct in6_addr *) (v)))
        ^
2 warnings generated.
```

This is probably just because of [the space deleted by that PR](https://github.com/ocsigen/lwt/pull/416/files#diff-e7b477b8db74e783e0d540162677ec4fR839).

However, instead of reintroducing the space, I think we can delete these macro definitions. They were [added to Lwt](https://github.com/ocsigen/lwt/commit/d91e1f55f4539a1d4e8d58afe631b64b5b44c11d#diff-32dd5e49710c54ddd1d5c20fd2d11e79R2290) in 2010, and `caml/socketaddr.h` distributed with OCaml was [included](https://github.com/ocsigen/lwt/commit/155905ca7db15949801c30774042ec0ad573bfe9#diff-1ec0034c530cf0901cf4011f943aacdbR31) seven months ago, at which point the macros silently shadowed each other. We don't need the Lwt instances.

See [CONTRIBUTING.md](https://github.com/ocsigen/lwt/blob/master/doc/CONTRIBUTING.md#contributing-to-the-lwt-code) for the code workflow.

**Comment by raphael-proust** [comment]
Other warnings:

```
lwt_unix_unix.h: In function ‘worker_readdir’:
lwt_unix_unix.h:1787:5: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
     job->result = readdir_r(job->dir, job->entry, &job->ptr);
     ^~~
In file included from /usr/include/features.h:410:0,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/stdint.h:9,
                 from /usr/lib/ocaml/caml/config.h:99,
                 from /usr/lib/ocaml/caml/misc.h:24,
                 from /usr/lib/ocaml/caml/alloc.h:23,
                 from lwt_unix_stubs.c:29:
/usr/include/dirent.h:189:12: note: declared here
 extern int __REDIRECT (readdir_r,
            ^
In file included from lwt_unix_stubs.c:98:0:
lwt_unix_unix.h: In function ‘worker_readdir_n’:
lwt_unix_unix.h:1833:9: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
         int result = readdir_r(job->dir, &job->entries[i], &ptr);
         ^~~
In file included from /usr/include/features.h:410:0,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/include/stdint.h:9,
                 from /usr/lib/ocaml/caml/config.h:99,
                 from /usr/lib/ocaml/caml/misc.h:24,
                 from /usr/lib/ocaml/caml/alloc.h:23,
                 from lwt_unix_stubs.c:29:
/usr/include/dirent.h:189:12: note: declared here
 extern int __REDIRECT (readdir_r,
            ^
```

```
File "_none_", line 1:
Warning 58: no cmx file was found in path for module Toploop, and its interface was not compiled with -opaque
```

**Comment by aantron** [comment]
Thanks for taking care of this. I appreciate the thoroughness.

Some notes about warnings:
- There are also some warnings on Windows, e.g. https://ci.appveyor.com/project/aantron/lwt/build/531/job/d2jfk3t5uk6nqj3d#L822. I'm not exactly suggesting you fix those too, but if you want to give it a shot, glance in the AppVeyor logs to see the current state.
- We can't squash warning 58 except by disabling it, as that is due to the packaging of other libraries. We generally had it disabled before, so maybe we need to restore that.
- Before the Jbuilder port, we compiled with all OCaml warnings as errors inside CI (only). I believe this isn't the case anymore. We should probably open a separate issue for restoring that (cc @andrewray).
- I also have a longstanding dream of compiling with C warnings as errors inside CI. I had made a push (5d0cf7910c547482607d2a3986296b173876fd78 and surrounding commits) to get rid of warnings in the past, but never got around to fully cleaning the C build.
